modules/nodes-sigstore-using-validation.adoc

```diff
@@ -0,0 +1,127 @@
+// Module included in the following assemblies:
+//
+// * nodes/nodes-sigstore-using.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="nodes-sigstore-using-validation_{context}"]
+= Validate cluster-wide policies using sigstore
+
+The following procedure demonstrates how to validate cluster-wide policies using Sigstore in {product-title}.
+
+.Prerequisites
+
+* Ensure that the policy on OpenShift image repositories (quay.io/openshift-release-dev/ocp-release, quay.io/openshift-release-dev/ocp-v4.0-art-dev) allows deployment of OpenShift images required for cluster operation.
+
+.Procedure
+
+. Build and Push an Image for Signing:
++
+[source,sh]
+----
+podman login quay.io
+Enter the username and password as prompted:
+Username: qiwanredhat
+Password: <your-password>
+Login Succeeded!
+----
+
+. Build and push the image:
++
+[source,sh]
+----
+echo "FROM scratch" > Dockerfile
+podman build -t quay.io/qiwanredhat/test-sigstoresigned .
+podman push quay.io/qiwanredhat/test-sigstoresigned:latest
+----
+
+. Generate a key pair for cosign:
++
+[source,sh]
+----
+cosign generate-key-pair
+Enter a password for the private key when prompted. The keys will be saved as cosign.key and cosign.pub.
+Sign the Image Using Cosign:
+Sign the image using your private key and push the signature to the registry:
+sh
+cosign sign --key ./cosign.key --registry-username=<your-username> --registry-password=<your-password> quay.io/qiwanredhat/test-sigstoresigned:latest
```

Comment:
Push + later sign is discouraged, this is vulnerable to attacks by the malicious registry.

Use `podman push --sign-by-sigstore-private-key` or `--sign-by-sigstore` to do this securely for just-built images.

(And for the use cases where `cosign sign` were truly required, use `--recursive` to interoperate with OCP, and IIRC `--sign-container-identity quay.io/qiwanredhat/test-sigstoresigned:latest` is typically required.)

Reply:
… I’d say that `cosign sign` should be _extremely rarely_ ”truly required”. `skopeo copy --sign…` to promote an image from staging to production and sign, for example.